### PR TITLE
chore: Remove need for insecure registry

### LINKED ISSUE #496: Precommit script fails locally
Whenever I run `./scripts/pre-commit.sh` locally, I get the following error:
```
Downloading test dependencies
Checking initial state of working tree
Verifying Gofmt
Please run hack/update-gofmt.sh to fix the following files:
./pkg/kubefed2/federate/federate.go
```

I plan to submit a PR that updates `federate.go` to be gofmt'd.

### PR DESCRIPTION
**Pull request by jimmidyson**
**What this PR does / why we need it**:

Removes the need for the insecure registry deployment and configuration from running e2e tests locally with kind.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #496.

**Special notes for your reviewer**:

### CODE REVIEW

**Review by makkes — CHANGES_REQUESTED** [review]
please update docs/environments/kind.md accordingly.

**Comment by jimmidyson** [comment]
@makkes Done.

File: scripts/deploy-kubefed.sh [hunk 1]
```diff
@@ -153,19 +155,14 @@ check-command-installed helm
 # Build KubeFed binaries and image
 if [[ "${USE_LATEST:-}" != "y" ]]; then
   cd "$(dirname "$0")/.."
-  make container IMAGE_NAME=${IMAGE_NAME}
+  make container IMAGE_NAME="${IMAGE_NAME}"
   cd -
-  # Use DOCKER_PUSH=n ./scripts/deploy-kubefed.sh <image> to skip docker
-  # push on container image when not using latest image.
-  if [[ "${DOCKER_PUSH:-y}" == "y" ]]; then
```
Comment (makkes) [inline scripts/deploy-kubefed.sh:160]: This variable is still passed in from the [Makefile](https://github.com/kubernetes-sigs/kubefed/blob/59e60d97b5b79ff3b2dfe73bbecaa188938be372/Makefile#L169)

**Review by makkes — COMMENTED** [review]
I'm testing the changes locally now, will report my results shortly. Besides that, a small nit.

File: scripts/deploy-kubefed.sh (hunk 1 above)
Comment (jimmidyson) [reply scripts/deploy-kubefed.sh:160]: Removed, good catch!

**Comment by jimmidyson** [comment]
Hmm seems to be a consistent test failure.

**Review by makkes — APPROVED** [review]
macOS still fails, let's delay fixing that to another PR. Tested on Linux now and it works fine.

**Pull request merged**